app/src/main/java/com/cmput301f20t21/bookfriends/repositories/UserRepository.java

```diff
@@ -51,23 +55,41 @@ public static IUserRepository getInstance() {
         });
     }
 
-    public Task<DocumentSnapshot> getByUid(String uid) {
-        return userCollection.document(uid).get();
+    public Task<User> getByUid(String uid) {
+        return userCollection.document(uid)
+                .get()
+                .continueWith(task -> {
+                    if (task.getResult() != null) {
+                        return task.getResult().toObject(User.class);
+                    }
+                    throw new UserNotExistException();
+                });
     }
 
-    public Task<QuerySnapshot> getByUsernameStartWith(String username) {
+    public Task<List<User>> getByUsernameStartWith(String username) {
+        final Query userQuery;
         if (username.length() == 0) {
-            // because username is alphanumeric, according to ascii table, @ is less than 'a' and 0
```

Comment:
maybe keep the comment here, it's hard to understand the logic here without the comment